Add WorldObjectService, SpotCheckService, unit tests, Docker CI build pipeline

jeremysee2 opened the pull request:
The corresponding change in `spot_wrapper` is at: https://github.com/bdaiinstitute/spot_wrapper/pull/6

jeremysee2 commented:
Reference to #104 and corresponding PR's in the Spot Wrapper repo at https://github.com/bdaiinstitute/spot_wrapper/pull/6

This handles the breaking changes introduced in the updates to the spot_wrapper, and adds some new features:

1. WorldObject service triggers in a AsyncPeriodicQuery, updating as a custom ROS message and publishing relevant TF's 
2. SpotCheck service is made available, to trigger joint calibration from ROS
3. GraphNav has been refactored; usage will be using `navigate_init`, `navigate_to` and `navigate_route`, with services to upload and download graphs to/from local storage
4. Testing coverage for ROS helper functions, and basic ROS-level unit tests for the wrapper
5. CI pipeline to automatically run tests, although they don't say anything about the correctness in Github Actions, needs manual inspection to see if they are passing
6. CI pipeline to build a Docker container for deployment on Spot, just requires setting ENV variables for username and password

Documentation to be added soon.

jeremysee2 commented:
For the Docker build pipeline, would require the maintainer to add Docker Hub secrets through Github; on my fork it's using my Docker hub account at https://hub.docker.com/r/jeremysee/spot_ros

heuristicus commented on spot_driver/src/spot_driver/ros_helpers.py:
```diff
@@ -434,7 +468,71 @@ def get_corrected_odom(base_odometry: Odometry):
 
 
-def GetOdomFromState(state, spot_wrapper, use_vision=True):
+def get_corrected_odom(base_odometry: Odometry):
```
It looks like this function got duplicated somewhere along the way, probably in a merge.

jeremysee2 replied:
Resolved in [59b2db4](https://github.com/heuristicus/spot_ros/pull/126/commits/59b2db42a21371f197d8a0f0bbba65a26ce96a47)

jeremysee2 commented:
Added docs for Arm, EAP, GraphNav and Docker.

heuristicus reviewed (commented):
Thanks for your work on this. I think the main thing to ask is that you add some comments to functions that you've added. The driver in general is fairly sparse on comments in the code and I want to start improving that. The same applies to messages and services, which are not particularly well commented either.

I still need to:
- [x] test it on the robot in conjunction with the wrapper changes
- [ ] check docker stuff

jeremysee2 commented:
Added the requested comments, fixed `ros_helpers_test.py` (https://github.com/jeremysee2/spot_ros/blob/master/spot_driver/test/ros_helpers_test.py#L545), reverted the default `hostname` param in ad7687f
